test/e2e/specs/editor/blocks/columns.spec.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	getBlockFunction: async ( { page }, use ) => {
+		await use( new GetAllBlockInserterItem( { page } ) );
+	},
+} );
+
+test.describe( 'Columns Test', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test( 'Create Columns and verify', async ( {
+		page,
+		editor,
+		getBlockFunction,
+	} ) => {
+		// Open Columns
+		await editor.insertBlock( { name: 'core/columns' } );
+		await page.locator( '[aria-label="Two columns; equal split"]' ).click();
+
+		// Open List view toggle
+		await page.locator( 'role=button[name="List View"i]' ).click();
+
+		await getBlockFunction.get_list_view_block( 'Column' );
+
+		// Toggle Block inserter
+		await page
+			.locator( 'role=button[name="Toggle block inserter"i]' )
+			.click();
+
+		// Verify Column
+		expect( await getBlockFunction.get_list_all_block() ).toHaveLength( 1 );
+	} );
+} );
+
+class GetAllBlockInserterItem {
+	constructor( { page } ) {
+		this.page = page;
+	}
+
+	async get_list_view_block( blockLabel ) {
+		return this.page
+			.locator(
+				`//table[contains(@aria-label,'Block navigation structure')]//a[.//span[text()='${ blockLabel }']]`
+			)
+			.first()
+			.click();
+	}
+	async get_list_all_block() {
+		await this.page.waitForTimeout( 500 );
+		const inserterItemTitles = await this.page.evaluate( () => {
+			return Array.from(
+				document.querySelectorAll(
+					'.block-editor-block-types-list__item-title'
+				)
+			).map( ( inserterItem ) => {
+				return inserterItem.innerText;
+			} );
+		} );
+		return [ ...new Set( inserterItemTitles ) ].sort();
+	}
+}
```

Comment:
I don't think we need helper utils for a singular test. It's probably better if we inline them. What do you think?

The e2e test docs also [recommend using role selectors](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#best-practices). Would it be possible to rewrite these assertions using role selectors?

The Chrome DevTools now has an accessibility tree inspector, so you can use it to find the correct roles for the elements. See https://developer.chrome.com/blog/full-accessibility-tree/.

Reply:
@Mamaduka  Okay  I will try to go with the inline approach and will try to use a role selector instead of others

Reply:
@Mamaduka I tried to follow the inline approach. I implemented that also but I was not able to execute document query function with role selector. Thats why used the CSS selector for evaluating.